app/models/order_detail.rb

```diff
@@ -735,8 +742,8 @@ def cancel_reservation(canceled_by, order_status: OrderStatus.canceled_status, a
       end
     else
       return false unless res && res.can_cancel?
-      res.canceled_at = Time.zone.now # must set canceled_at after calling #can_cancel?
-      return false unless res.save
+      self.canceled_at = Time.zone.now # must set canceled_at after calling #can_cancel?
+      return false unless save
```

Comment:
Could these two lines now be rolled up into something like `return false unless update(canceled_at: Time.current)`?